josue/js/app-test.js

```diff
@@ -0,0 +1,159 @@
+/** JAVASCRIPT REDUCED IN ONE FILE TO ADD IT WHEN TESTING */
+/**NO NEED TO READ IT, IT'S THE SAME CODE */
+class FormValidator{
+    constructor(form,fields){
+        this.form = form
+        this.fields = fields
+    }
+    initialize(){
+        this.validateOnEntry()
+    }
+    validateOnSubmit(){
+        this.fields.map(field => {
+            const input = document.querySelector(`#${field}`);
+            this.validateFields(input);
+        });
+        //If all the inputs have the class input--correct the function returns true 
+        const correctInputs = this.fields.filter(field =>{
+            //Check in the DOM if the field has the class input--correct
+            const input = document.querySelector(`#${field}`);
+            if(input.classList.contains('input--correct')){
+                return field
+            }
+        });
+        if(correctInputs.length == this.fields.length){
```

Comment:
this entire `if` block can be resumed in `return correctInputs.length === this.fields.length;` and always use `===` instead of `==`